scripts/create-install-iso.sh

```diff
@@ -0,0 +1,262 @@
+
+# bootloader config files etc. - like "cp -r *", not forgetting .treeinfo
+tar -C "$topdir/iso/$DIST" -cf - . | tar -C "$ISODIR/" -xf - ${VERBOSE}
+
+# initrd
+cp ${VERBOSE} -a "$INSTALLIMG" $ISODIR/install.img
+
+# kernel from rpm
+get_rpms "$SCRATCHDIR" kernel
+rpm2cpio $SCRATCHDIR/kernel-*.rpm | (cd $ISODIR && cpio ${VERBOSE} -idm "*vmlinuz*")
+rm ${VERBOSE} $ISODIR/boot/vmlinuz-*-xen
+mv ${VERBOSE} $ISODIR/boot/vmlinuz-* $ISODIR/boot/vmlinuz
+
+# alt kernel from rpm
+get_rpms "$SCRATCHDIR" kernel-alt
+rpm2cpio $SCRATCHDIR/kernel-alt-*.rpm | (cd $ISODIR && cpio ${VERBOSE} -idm "*vmlinuz*")
+rm ${VERBOSE} $ISODIR/boot/vmlinuz-*-xen
+mkdir ${VERBOSE} $ISODIR/boot/alt
+mv ${VERBOSE} $ISODIR/boot/vmlinuz-* $ISODIR/boot/alt/vmlinuz
+
+# xen from rpm
+get_rpms "$SCRATCHDIR" xen-hypervisor
+rpm2cpio $SCRATCHDIR/xen-hypervisor-*.rpm | (cd $ISODIR && cpio ${VERBOSE} -idm "*xen*gz")
+mv ${VERBOSE} $ISODIR/boot/xen-*-d.gz $ISODIR/boot/xen.gz
+rm ${VERBOSE} $ISODIR/boot/xen-*.gz
+
+
+# Memtest86
+get_rpms "$SCRATCHDIR" memtest86+
+rpm2cpio $SCRATCHDIR/memtest86+-*.rpm | (cd $ISODIR && cpio ${VERBOSE} -idm "./boot/*")
+if [ ! -r $ISODIR/boot/memtest.bin ]; then
+    # older 5.x packaging
+    mv ${VERBOSE} $ISODIR/boot/memtest86+-* $ISODIR/boot/memtest.bin
+fi
+
+
+# optional local repo
+
+if [ $DOREPO = 1 ]; then
+    mkdir ${VERBOSE} "$ISODIR/Packages"
+
+    get_rpms --depends "$ISODIR/Packages" xcp-ng-deps kernel-alt
+
+    createrepo_c ${VERBOSE} "$ISODIR"
+    if [ -n "$KEYID" ]; then
+        gpg1 ${VERBOSE} --default-key="$KEYID" --armor --detach-sign "$ISODIR/repodata/repomd.xml"
+	gpg1 ${VERBOSE} --armor -o "$ISODIR/RPM-GPG-KEY-$KEYNICK" --export "$KEYID"
+	[ -z "$VERBOSE" ] || echo "using key RPM-GPG-KEY-$KEYNICK in .treeinfo"
+	sed -i "s,key1 = .*,key1 = RPM-GPG-KEY-$KEYNICK," \
+	    $ISODIR/.treeinfo
+    else
+	# installer checks if keys are here even when verification is disabled
+	[ -z "$VERBOSE" ] || echo "disabling keys in .treeinfo"
+	sed -i "s,^key,#key," \
+	    $ISODIR/.treeinfo
+
+	# don't try to validate repo sig if we put none
+	[ -z "$VERBOSE" ] || echo "adding no-repo-gpgcheck to boot/isolinux/isolinux.cfg boot/grub/grub*.cfg"
+	sed -i "s,/vmlinuz,/vmlinuz no-repo-gpgcheck," \
+	    $ISODIR/boot/isolinux/isolinux.cfg \
+	    $ISODIR/boot/grub/grub*.cfg
+    fi
+else
+    # no repo
+    # FIXME: should be generated above instead?
+    rm ${VERBOSE} "$ISODIR/.treeinfo"
+
+    # FIXME: trigger netinstall mode?
+fi
+
+
+# BIOS bootloader: isolinux from rpm
+
+get_rpms "$SCRATCHDIR" syslinux
+mkdir "$SCRATCHDIR/syslinux"
+rpm2cpio $SCRATCHDIR/syslinux-*.rpm | (cd "$SCRATCHDIR/syslinux" && cpio ${VERBOSE} -idm "./usr/share/syslinux/*")
+
+cp ${VERBOSE} -p \
+   "$SCRATCHDIR/syslinux/usr/share/syslinux/isolinux.bin" \
+   "$SCRATCHDIR/syslinux/usr/share/syslinux/mboot.c32" \
+   "$SCRATCHDIR/syslinux/usr/share/syslinux/menu.c32" \
+   \
+   $ISODIR/boot/isolinux/
+
+
+## create final ISO
+
+if false; then
```

Comment:
Why?

Reply:
grub-mkrescue is "the reference" in someway, providing largest platform support for booting, but OTOH adds tons of stuff we don't need at all. It was used as a reference to implement UEFI boot, and I kept it to have it handy if/when we need it, since the command options are not that obvious. Eg. we may want to add support for x86 Macs some day.
Will add a comment.